e2e/rdctl.e2e.spec.ts

```diff
@@ -376,11 +376,45 @@ test.describe('Command server', () => {
     });
   });
 
-  test('should not update values when the /transient_settings payload has errors', async() => {
+  test('should not update values when the /transient_settings currentNavItem payload is invalid', async() => {
     let resp = await doRequest('/v0/transient_settings');
     const transientSettings = await resp.json();
 
-    const requestedSettings = _.merge({}, transientSettings, { preferences: { currentNavItem: 'foo' } });
+    const requestedSettings = _.merge({}, transientSettings, { preferences: { currentNavItem: { name: 'foo', bar: 'bar' } } });
+    const resp2 = await doRequest('/v0/transient_settings', JSON.stringify(requestedSettings), 'PUT');
+
+    expect(resp2.ok).toBeFalsy();
+    expect(resp2.status).toEqual(400);
+
+    // Now verify that the specified values did not get updated.
+    resp = await doRequest('/v0/transient_settings');
+    const refreshedSettings = await resp.json();
+
+    expect(refreshedSettings).toEqual(transientSettings);
+  });
+
+  test('should not update values when the /transient_settings payload has invalid currentNavItem name', async() => {
+    let resp = await doRequest('/v0/transient_settings');
+    const transientSettings = await resp.json();
+
+    const requestedSettings = _.merge({}, transientSettings, { preferences: { currentNavItem: { name: 'foo' } } });
+    const resp2 = await doRequest('/v0/transient_settings', JSON.stringify(requestedSettings), 'PUT');
+
+    expect(resp2.ok).toBeFalsy();
+    expect(resp2.status).toEqual(400);
+
+    // Now verify that the specified values did not get updated.
+    resp = await doRequest('/v0/transient_settings');
+    const refreshedSettings = await resp.json();
+
+    expect(refreshedSettings).toEqual(transientSettings);
+  });
+
```

Comment:
I would add tests to verify that you can't update transient settings when you specify a tab for a setting that doesn't support tabs: `'Container Engine'`, `'Kubernetes'`, and `win32 ? 'WSL' : 'Virtual Machine'`

Reply:
Sure, thanks Eric, just pushed.